### PR TITLE
Add Java 15 to CI

### PR DESCRIPTION
**Pull request by martint**

### CODE REVIEW

**Comment by findepi** [comment]
Annotation's tostring changed.

File: discovery/src/test/java/io/airlift/discovery/client/TestServiceTypes.java [hunk 1]
```diff
@@ -50,17 +49,6 @@ public void testValue()
         assertEquals(ServiceTypes.serviceType("type").value(), "type");
     }
 
-    @Test
-    public void testToString()
```
Comment (martint) [inline discovery/src/test/java/io/airlift/discovery/client/TestServiceTypes.java:54]: Java doesn't make any guarantees about the format of an annotation's toString and it seems to change from version to version, so I chose to remove the test altogether.

Comment (findepi) [reply discovery/src/test/java/io/airlift/discovery/client/TestServiceTypes.java:54]: rel https://github.com/google/guice/issues/1312

we do not need to make the toString match JVM provided one, but it's still valuable to test toString is something reasonable (fixed const value)

**Review by findepi — APPROVED** [review]

File: discovery/src/test/java/io/airlift/discovery/client/TestServiceTypes.java (hunk 1 above)
Comment (electrum) [reply discovery/src/test/java/io/airlift/discovery/client/TestServiceTypes.java:54]: I merged this so we can move forward with testing Java 15 in the CI. I filed #902 for follow up on this.

**Pull request merged**